docs/examples/api-client.ts

```diff
@@ -384,30 +441,119 @@ export class TradingSimulatorClient {
     return this.request<any>('POST', '/api/trade/execute', payload);
   }
 
+  /**
+   * Get a quote for a potential trade between two tokens
+   * 
+   * @param params Trade quote parameters
+   * @returns A promise resolving to the quote response
+   */
+  async getQuote(params: {
+    fromToken: string;
+    toToken: string;
+    amount: string;
+    fromChain?: BlockchainType;
+    toChain?: BlockchainType;
+    fromSpecificChain?: SpecificChain;
+    toSpecificChain?: SpecificChain;
+  }): Promise<{
+    fromToken: string;
+    toToken: string;
+    fromAmount: number;
+    toAmount: number;
+    exchangeRate: number;
+    slippage: number;
+    prices: {
+      fromToken: number;
+      toToken: number;
+    };
+    chains: {
+      fromChain: string;
+      toChain: string;
+    };
+  }> {
+    // Build the query string
+    let query = `?fromToken=${encodeURIComponent(params.fromToken)}&toToken=${encodeURIComponent(params.toToken)}&amount=${encodeURIComponent(params.amount)}`;
+    
+    // Add optional parameters if they exist
+    if (params.fromChain) query += `&fromChain=${params.fromChain}`;
+    if (params.toChain) query += `&toChain=${params.toChain}`;
+    if (params.fromSpecificChain) query += `&fromSpecificChain=${params.fromSpecificChain}`;
+    if (params.toSpecificChain) query += `&toSpecificChain=${params.toSpecificChain}`;
+    
+    return this.request<any>('GET', `/api/trade/quote${query}`);
+  }
+
   /**
    * Get the status of the current competition
    * 
    * @returns A promise that resolves to the competition status response
    */
-  async getCompetitionStatus(): Promise<any> {
+  async getCompetitionStatus(): Promise<{
```

Comment:
It might be good to pull out all these response types so that they can be used on the caller side, and to avoid all the self.request<any> stuff.

Reply:
resolved in [4e3e7ea](https://github.com/recallnet/trading-simulator/pull/3/commits/4e3e7eac04b1d3373cfb2c666826cb69e73adfe8)